RA-17

KinCadeFoster opened the pull request:
- Добавил логику сохранения состояния с SharedPreferences.
- Добавил логику обработки состояния из SharedPreferences и установку соответсвующего состояния на краточке рецепта

KinCadeFoster committed:
RA-17
- Добавил логику сохранения состояния с SharedPreferences.
- Добавил логику обработки состояния из SharedPreferences и установку соответсвующего состояния на краточке рецепта

aGroupoid commented on app/src/main/java/ru/example/recipeapp/RecipeFragment.kt:
```diff
@@ -70,16 +76,34 @@ class RecipeFragment : Fragment() {
         }
 
+        val favorites = getFavorites()
+        isFavorite = favorites.contains(recipe.id.toString())
```
Выглядит, что эта переменная лишняя. Перед её использованием ты всегда меняешь её значение на новое, может тогда делать это через локальную переменную?

aGroupoid commented on app/src/main/java/ru/example/recipeapp/RecipeFragment.kt:
```diff
@@ -22,6 +24,10 @@ class RecipeFragment : Fragment() {
     private var isFavorite: Boolean = false
     private lateinit var ingredientsAdapter: IngredientsAdapter
```
lateinit - заманчивая конструкция, но опасная и никак не спасающая от NullPointerException. 
Лучше стараться продумать подходящее поведение по-умолчанию. В уроке я реализовывал backing property (резервное свойство). Но здесь предлагаю применить более элегантный вариант с помощью делегата lazy, который использует ленивую инициализацию. Это такой механизм, который в лямбде принимает в нашем случае inflate() и возвращает созданный объект, который будет создан при первом обращении к нему.

KinCadeFoster replied:
у нас же доступа к recipe нет, пока вижу один враиант, передавать в конструктор фрагмента.

aGroupoid replied:
В твоём случае лучше тогда просто проинициализировать адаптер сразу с пустым списком. Потом менять его dataSet с актуальным списком в onCreate, если обновление не будет происходить, надо уведомить адаптер об изменениях и вызвать adapter.notifyDataSetChanged().

В конструктор фрагментов ничего передавать не надо, после поворота экрана, фрагмент пересоздастся и все данные из конструктора сотрутся. Ссылаться будет не на что

aGroupoid reviewed (changes requested):
Молодец, по текущему заданию сделал всё верно! Только щас заметил, что у ингредиентов нету величины измерения как на макете фигма. Поправь в этом ПР эту проблему

KinCadeFoster committed:
RA-17
- Правки по замечаниям
- Попраивл отображение едениц измеренеия у вывода рецепта

aGroupoid reviewed (approved):
Апрувлю,  поправь в следующем ПР коммент. Еще заметил,  при входе на экран Рецептов - количество порций пустое.  Добавь начальное значение для этой textView
